tools/structure/src/main.py

```diff
@@ -242,6 +252,21 @@ def run(
             structured_output_dict[SettingsKeys.METADATA] = metadata
             structured_output = json.dumps(structured_output_dict)
 
+        metrics = structured_output_dict[SettingsKeys.METRICS]
+        new_metrics = {}
+        if tool_settings[SettingsKeys.ENABLE_SINGLE_PASS_EXTRACTION]:
+            new_metrics = {
+                **metrics,
+                **index_metrics,
+            }
+        else:
+            # Merge dictionaries
+            new_metrics = {
+                key: {**metrics.get(key, {}), **index_metrics.get(key, {})}
+                for key in set(metrics)
+                | set(index_metrics)  # Union of keys from both dictionaries
+            }
+        structured_output_dict[SettingsKeys.METRICS] = new_metrics
```

Comment:
NIT: Suggest to add this as a separate function to reduce the growing size of this method

Reply:
Will handle this in upcoming PR.